package/src/components/Message/MessageSimple/MessageContent.tsx

```diff
@@ -440,21 +440,28 @@ const areEqual = <StreamChatGenerics extends DefaultStreamChatGenerics = Default
   const quotedMessageEqual =
     prevMessage.quoted_message?.id === nextMessage.quoted_message?.id &&
     isPrevQuotedMessageTypeDeleted === isNextQuotedMessageTypeDeleted;
-
   if (!quotedMessageEqual) return false;
 
-  const prevAttachments = prevMessage.attachments;
-  const nextAttachments = nextMessage.attachments;
+  const prevMessageAttachments = prevMessage.attachments;
+  const nextMessageAttachments = nextMessage.attachments;
   const attachmentsEqual =
-    Array.isArray(prevAttachments) && Array.isArray(nextAttachments)
-      ? prevAttachments.length === nextAttachments.length &&
-        prevAttachments.every(
-          (attachment, index) =>
-            attachment.image_url === nextAttachments[index].image_url &&
-            attachment.og_scrape_url === nextAttachments[index].og_scrape_url &&
-            attachment.thumb_url === nextAttachments[index].thumb_url,
-        )
-      : prevAttachments === nextAttachments;
+    Array.isArray(prevMessageAttachments) &&
+    Array.isArray(nextMessageAttachments) &&
+    prevMessageAttachments.length === nextMessageAttachments.length &&
+    prevMessageAttachments.every((attachment, index) => {
+      const attachmentKeysEqual =
+        attachment.type === 'image'
+          ? attachment.image_url === nextMessageAttachments[index].image_url &&
+            attachment.thumb_url === nextMessageAttachments[index].thumb_url
+          : attachment.type === nextMessageAttachments[index].type;
+
+      if (isAttachmentEqual)
+        return (
+          attachmentKeysEqual && !!isAttachmentEqual(attachment, nextMessageAttachments[index])
+        );
+
+      return attachmentKeysEqual;
+    });
```

Comment:
@khushal87 why did we change the logic here? :(

where is `attachment.type === 'image'` coming from?

Reply:
This broke the giphy shuffle behaviour ... and idk what not!!

Reply:
Even I am not sure how it came, I remember reshuffling the code to a variable but not this, anyways looking into it, gonna fix it soon.